WordPress/src/main/java/org/wordpress/android/ui/reader/services/post/ReaderPostLogicFactory.kt

```diff
@@ -0,0 +1,14 @@
+package org.wordpress.android.ui.reader.services.post
+
+import org.wordpress.android.ui.reader.repository.ReaderPostRepository
+import org.wordpress.android.ui.reader.services.ServiceCompletionListener
+import javax.inject.Inject
+
+class ReaderPostLogicFactory @Inject constructor(
+    private val readerPostRepository: ReaderPostRepository,
+) {
+    fun create(listener: ServiceCompletionListener) = ReaderPostLogic(
```

Comment:
I'm not sure, when the expression function directly calls the constructor I feel it's too redundant to add the return type, for example, this line would become:

```kotlin
fun create(listener: ServiceCompletionListener): ReaderPostLogic = ReaderPostLogic(
```

I find it a bit too verbose, but I can change it to that if it's better.

Reply:
I remember seeing something about it in the Kotlin coding conventions, but the only thing I could find was this regarding libraries:

<img width="754" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/1c9b9820-c6a9-48cb-b9c8-161d6018cda2">

Besides the obvious point above for libraries, I believe it helps to have the return type to help whoever is maintaining the code/debugging quickly identify the return type, but since for now this function only calls the constructor I don't see any significant improvement. Let's leave it as it is.